include/rnp/rnp2.h

```diff
@@ -260,6 +264,54 @@ rnp_result_t rnp_key_have_public(rnp_key_handle_t key, bool *result);
 
 /* TODO define functions for password-based encryption */
 
+rnp_result_t rnp_op_sign_create(rnp_op_sign_t *op,
+                                rnp_ffi_t      ffi,
+                                rnp_input_t    input,
+                                rnp_output_t   output);
+
+rnp_result_t rnp_op_sign_set_signing_key(rnp_op_sign_t op, rnp_key_handle_t key);
+
+rnp_result_t rnp_op_sign_set_compression(rnp_op_sign_t op, const char *compression, int level);
+rnp_result_t rnp_op_sign_set_armor(rnp_op_sign_t op, bool armored, bool cleansign);
```

Comment:
typo: cleansign -> clearsign

Reply:
I would vote for separation between armoring and cleartext signing. First is applicable for both encryption and signing of binary/text data, while cleartext signing is applicable only for signing of text documents.

Reply:
I agree with @ni4 here.

Reply:
Fixed - now 2 distinct functions for these